pkg/standalone-utils/contracts/test/MockEulerToken.sol

```diff
@@ -0,0 +1,108 @@
+    // in order to invest an asset to earn interest, you need to `deposit` into an eToken
+
+    uint256 public exchangeRateMultiplier;
+
+    // solhint-disable-next-line var-name-mixedcase
+    address public immutable ASSET;
+    IMockEulerProtocol public immutable EULER_PROTOCOL;
+
+    uint256 private _onlyForMockStorage;
+    address private _mockMsgSender;
+
+    constructor(
+        string memory name,
+        string memory symbol,
+        uint8 decimals,
+        address asset,
+        IMockEulerProtocol eulerProtocol
+    ) TestToken(name, symbol, decimals) {
+        exchangeRateMultiplier = 1;
+        ASSET = asset;
+        EULER_PROTOCOL = eulerProtocol;
+    }
+
+    // convert eToken Balance to underlying
+    function convertBalanceToUnderlying(uint256 balance) external override view returns (uint256) {
+        return balance * exchangeRateMultiplier;
+    }
+
+    function setExchangeRateMultiplier(uint256 _exchangeRateMultiplier) external {
+        require(_exchangeRateMultiplier < 3, "Cannot set exchangeRateMultiplier bigger 3");
+        exchangeRateMultiplier = _exchangeRateMultiplier;
+    }
+
+    // convert underlying Balance to eToken
+    function convertUnderlyingToBalance(uint256 balance) external override view returns (uint256) {
+        return balance * (1/exchangeRateMultiplier);
+    }
+
+    function underlyingAsset() external override view returns (address) {
+        return ASSET;
+    }
+
+    function deposit(uint256 subAccountId, uint256 amount) external override {
+
+        _onlyForMockStorage = subAccountId;
+        // Relayer only uses one account. Meaning no subAccountID required
+        // is set to 0 in the Relayer
+        // Transfer underlying tokens from sender to the Euler pool, and increase account's eTokens
+
+        // assumes Euler protocol has approval to move `amount` from users wallet
+        // mints MockEulerTokens to msg.sender (which is the relayer)
+        // the relayer has earlier used `_pullToken` to ensure it is the contract
+        // which is calling `deposit` on an eToken.
+
+        _mockMsgSender = msg.sender;
+
+        EULER_PROTOCOL.requestUnderlyingFromRelayer(address(ASSET), amount, _mockMsgSender); 
+
+        _mint(msg.sender, amount);
+    }
+
+    function withdraw(uint256 subAccountId, uint256 amount) external override {
+
+        _onlyForMockStorage = subAccountId;
+
+        // EulerWrapping will pass MAX_UINT as this allows to exchange all wrappedTokens
+        // for maximum amount of mainTokens
+        // this mock transforms the amount to be the balanceOf mockEulerTokens in the relayer
+        // so that the appropriate amount can be transferedFrom via the mockEulerProtocol
+        amount = this.balanceOf(msg.sender);
+
+        // request tokens from Euler protocol and send to the Relayer
+        _mockMsgSender = msg.sender;
+        EULER_PROTOCOL.sendUnderlyingToRelayer(ASSET, amount, _mockMsgSender);
+        _burn(msg.sender, amount);
```

Comment:
This isn't the correct amount to burn unless `exchangeRateMultiplier` is 1. We should use `convertBalanceToUnderlying` here.

Reply:
I misspoke here; it should have said `convertUnderlyingToBalance`. In any case, it's addressed now!